test/integration/protocol_integration_test.cc

```diff
@@ -248,6 +248,8 @@ TEST_P(ProtocolIntegrationTest, DrainClose) {
   EXPECT_EQ("200", response->headers().Status()->value().getStringView());
   if (downstream_protocol_ == Http::CodecClient::Type::HTTP2) {
     EXPECT_TRUE(codec_client_->sawGoAway());
+  } else {
+    EXPECT_EQ("close", response->headers().Connection()->value().getStringView());
```

Comment:
This is orthogonal, but noticed it doing test debugging when I was trying to tweak behavior, and I think we should have in there to make sure we don't simply time out.